library/core/src/ptr/non_null.rs

```diff
@@ -1829,4 +1854,59 @@ mod verify {
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+  
+    #[kani::proof_for_contract(NonNull::byte_sub)]
+    pub fn non_null_check_byte_sub() {
+        const SIZE: usize = mem::size_of::<i32>() * 100;
+        let mut generator = PointerGenerator::<SIZE>::new();
+        let count: usize = kani::any();
+        let raw_ptr: *mut i32 = generator.any_in_bounds().ptr as *mut i32;
+
+        unsafe {
+            let ptr = NonNull::new(raw_ptr).unwrap();
+            let result = ptr.byte_sub(count);
+        }
+    }
+
+  #[kani::proof_for_contract(NonNull::offset)]
+    pub fn non_null_check_offset(){
+        const SIZE: usize = mem::size_of::<i32>() * 100;
+        let mut generator1 = PointerGenerator::<SIZE>::new();
+        let mut generator2 = PointerGenerator::<SIZE>::new();
+        let start_ptr: *mut i32 = if kani::any() {
+            generator1.any_in_bounds().ptr as *mut i32
+        } else {
+            generator2.any_in_bounds().ptr as *mut i32
+        };
+        let ptr_nonnull = unsafe { NonNull::new(start_ptr).unwrap() };
+        let count: isize = kani::any();
+        unsafe { let result = ptr_nonnull.offset(count);
+       }
+}
+       
+    #[kani::proof_for_contract(NonNull::map_addr)]
+        pub fn non_null_check_map_addr() {
+        const SIZE: usize = mem::size_of::<i32>() * 100;
+        let mut generator = PointerGenerator::<SIZE>::new();
+        let ptr = generator.any_in_bounds().ptr as *mut i32;
+        let ptr_nonnull = unsafe { NonNull::new(ptr).unwrap() };
+        let new_offset: usize = kani::any_where(|&x| x < SIZE);
+        let f = |addr: NonZeroUsize| -> NonZeroUsize {
+        unsafe {NonZeroUsize::new_unchecked(addr.get().wrapping_add(new_offset)) 
+                } };
+        let result = ptr_nonnull.map_addr(f);
+}
+    #[kani::proof_for_contract(NonNull::with_addr)]
+    pub fn non_null_check_with_addr() {
+        const SIZE: usize = mem::size_of::<i32>() * 100;
+        let mut generator = PointerGenerator::<SIZE>::new();
+        let ptr = generator.any_in_bounds().ptr as *mut i32;
+        let ptr_nonnull = unsafe { NonNull::new(ptr).unwrap() };
+        let new_offset: usize = kani::any_where(|&x| x < SIZE);
+        let new_addr = unsafe {
+             NonZeroUsize::new_unchecked(ptr_nonnull.as_ptr() as usize + new_offset)
```

Comment:
This could cause UB because `ptr_nonnull` is a pointer to _somewhere_ in the generator's buffer, not the beginning, so adding `new_offset` to it could take you out of bounds (and overflow back to zero).
I would recommend making `ptr_nonnull` point to the beginning of an array, then calling `with_ptr` on the result of adding `new_offset` to that.